views/cypress/tests/test.spec.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA ;
+ */
+
+describe('Test', () => {
+    const testDirectoryName = 'Lorem ipsum dolar sit amet dir';
+    const testName = 'Lorem ipsum dolar sit amet name';
+
+    beforeEach(() => {
+        cy.fixture('urls').as('urls').then(urls => {
+            const username = Cypress.env('adminUser');
+            const password = Cypress.env('adminPass');
+
+            cy.login({ url: urls.login, username, password });
+        });
+    });
+
+    it('should reach the tests page', function () {
+        cy.visit(this.urls.root);
+
+        cy.contains('Tests').click();
+        cy.url().should('include', this.urls.tests);
+    });
+
+    it('should create class', function () {
+        cy.visit(this.urls.tests);
+        cy.contains('New class').click();
+
+        cy.contains('Edit class');
+        cy.contains('label', 'Label')
+            .parent()
+            .within(() => {
+                cy.get('input').clear().type(testDirectoryName);
+            });
+        cy.get('form button[type="submit"]').click();
+
+        cy.get('#tree-manage_tests')
```

Comment:
Repeated id could be extracted into a fixture file (similar to `fixtures/locators.json` in tao core).